shapeshift/web
# feat: improve thorchain status polling reliability

## Description

THOR Tx status detection can currently take a large amount of time - which is *not* a bug, outbound Txs sometimes take a very long time to get fired.
It looks like the API recently had some (at least as far as we're concerned) breaking changes where `outbound_signed` is always present, regardless of the THOR Tx type. As such, relying on its *presence*, or lack thereof, doesn't cut it anymore.
This PR introduces a `skipOutbound` option, which does precisely that, skips outbound signed checks for lending and savers Txs, since these do not rely on an outbound Tx.

Also includes some fixes/improvements for THORChain lending.

## Pull Request Type


## Issue (if applicable)

closes #5656

## Risk



## Testing



- Ensure depositing some collateral into lending spins until the deposit is complete, and the overview screen reflects the newly deposited amount once completed
- Ensure THORChain savers deposits/withdraws update to completed, including for UTXOs

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)

 
### Lending collateral deposit


https://github.com/shapeshift/web/assets/17035424/68284e06-6e48-4a86-9aff-17a754719373

### ATOM deposits

https://github.com/shapeshift/web/assets/17035424/56aec6c9-5e7e-4293-bf83-1f17e01be6f0

### ATOM withdraws


https://github.com/shapeshift/web/assets/17035424/0f2a63a0-9599-4578-89e4-e4c523656b84

### DOGE deposits

https://github.com/shapeshift/web/assets/17035424/781c2287-45dc-4ed6-9010-15fdb9253751

### DOGE withdraws

https://github.com/shapeshift/web/assets/17035424/25f1992f-79e8-4a01-9009-925fce002e76
